### ISSUE TITLE
"missing `#[package]` attribute" when deriving TryFromJavaValue but not Signature

### ISSUE THREAD

**Issue by kitlith** [giovanniberti/robusta#4]
I attempted a fix at https://github.com/kitlith/robusta/tree/fix_missing_package, which fixes the ostensible issue, but then realized that this "solution" is going to cause issues when you're not deriving Signature, because the classpath specified in Signature will not match the classpath generated for the derive macros.

I think the real/better solution here is to put the classpath in the signature trait as well as the signature itself, or some complication thereof. In my case, I cannot rely on the ident of the struct because i'm trying to work with an inner class `Outer$Inner`, and rust identifiers cannot contain '$'

inb4 there's another way you want to do this that I didn't think of.

**Comment by giovanniberti** [comment]
Sorry for the late reply.
I tried the approach you suggested and the fundamental problem is that when generating methods we have to produce an identifier for each function (e.g. `Java_com_bar_Quux_method1` for class `com.bar.Quux`). In order to do that via the `Signature` trait we would write `<Quux as Signature>::SIG_TYPE` and have to resolve that inside the macro code. 

Moreover, something like `fn const_format!("Java_{}", <Quux as Signature>::SIG_TYPE`) would not work as macros are not allowed in identifier position.

So I don't think that we can do autoderive without attributes.

**Comment by kitlith** [comment]
Ahhhh, okay. no problem!

**Comment by giovanniberti** [comment]
Btw, I'm thinking of adding a new attribute (and maybe remove `package`) to allow for subclasses. For the compiler error I think I'll just make that a warning and later see if something better can be done UX-wise.

**Comment by giovanniberti** [comment]
More on the issue: I [documented](https://github.com/giovanniberti/robusta/blob/13b330a4a721ed8d857c70b6d751da7d3c74315b/robusta-codegen/src/transformation/mod.rs#L211) this and promptly forgot until now :smile:

**Comment by kitlith** [comment]
@giovanniberti -- at a glance 904cbd5 appears to be a solution similar to my first attempt that I linked in my post opening the issue. I never PR'd it, because I realized that it would cause (harder to spot) issues with my actual goal, dealing with an Inner Class (or in general, classes with names not representable in rust source code). Should that be moved to a different issue, or is there a way to represent such a class name that I missed initially?
